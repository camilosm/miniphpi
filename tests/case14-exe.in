3
maria
pedro
paulo
7
5
8
