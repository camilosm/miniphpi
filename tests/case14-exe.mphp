/* Ler a quantidade */
$s = read "Entre com a quantidade de alunos: ";
if ($s < 0) {
    echo "Quantidade invalida\n";
} else {
    /* Ler os nomes dos alunos */
    $i = 0;
    $nomes = array();
    while ($i < $s) {
        $n = read "Entre com o nome do aluno " . ($i + 1) . ": ";
        if ($n == "") {
            echo "  Nome invalido\n";
        } else {
            $nomes[$i] = $n;
            $i++;
        }
    }

    /* Ler as notas dos alunos */
    $notas = array();
    foreach ($nomes as $id => $n) {
        $tmp = 0;
        while ($tmp == 0) {
            $x = read "Entre com a nota do aluno " . $n . ": ";
            if ($x >= 0 and $x <= 100) {
                $notas[$n] = $x;
                $tmp = 1;
            } else {
                echo "  Nota invalida\n";
            }
        }
    }

    /* Obter a menor e maior nota */
    $count = 0;
    $soma = 0;
    foreach ($notas as $n => $v) {
        if ($count++ == 0) {
            $menor = $n;
            $maior = $n;
        } else {
            if ($notas[$n] < $notas[$menor]) {
                $menor = $n;
            }

            if ($notas[$n] > $notas[$maior]) {
                $maior = $n;
            }
        }

        $soma += $v;
    }

    echo "\nA menor nota foi do " . $menor . " com " . $notas[$menor] . " pontos\n";
    echo "A maior nota foi do " . $maior . " com " . $notas[$maior] . " pontos\n";
    echo "Media das notas: " . ($soma / $count) . "\n";
}