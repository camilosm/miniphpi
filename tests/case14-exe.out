Entre com a quantidade de alunos: Entre com o nome do aluno 1: Entre com o nome do aluno 2: Entre com o nome do aluno 3: Entre com a nota do aluno maria: Entre com a nota do aluno pedro: Entre com a nota do aluno paulo:
A menor nota foi do pedro com 5 pontos
A maior nota foi do paulo com 8 pontos
Media das notas: 6